crates/test-framework/src/app_utils/mod.rs

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use std::borrow::Cow;
+use std::io::{self, Cursor};
+use std::path::PathBuf;
+
+use app::{App, AppBuilder};
+use spicepod::reader::ReadableYaml;
+use spicepod::{
+    Spicepod,
+    reader::{self, ReadablePath},
+};
+
+struct SpicepodString {
+    spicepod_str: String,
+}
+
+impl SpicepodString {
+    pub fn new(spicepod_str: Cow<'_, str>) -> Self {
+        Self {
+            spicepod_str: spicepod_str.into_owned(),
```

Comment:
Super nit: I'm curious why the `Cow`? It's fairly specific for the caller.

```suggestion
    pub fn new<S: ToString>(spicepod_str: S) -> Self {
        Self {
            spicepod_str: spicepod_str.to_string(),
```

Reply:
Ah yeah, this doesn't need to be a Cow anymore. My first implementation had tried to use the reference